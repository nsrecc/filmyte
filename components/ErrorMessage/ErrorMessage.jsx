import React from 'react';
import PropTypes from 'prop-types';
import styles from './ErrorMessage.module.scss';

const ErrorMessage = ({ message }) => {
  return (
    <div className={styles.errorMessageWrapper}>
      <h1>Error.</h1>
      <h2>{`${message && `${message}. `}Please try again later.`}</h2>
    </div>
  );
};

ErrorMessage.propTypes = {
  message: PropTypes.string,
};

ErrorMessage.defaultProps = {
  message: '',
};

export default ErrorMessage;
