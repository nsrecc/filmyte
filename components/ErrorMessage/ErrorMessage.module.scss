@use "styles/variables" as v;
@use "styles/mixins" as m;

// TO DO: mobile styling
.errorMessageWrapper {
  @include m.maxWidth;
  display: flex;
  justify-content: center;
  align-items: center;

  h1, h2 {
    padding: 0 2rem;

    &:first-child {
      border-right: 1px solid v.$gray-50;
    }
  }
}
