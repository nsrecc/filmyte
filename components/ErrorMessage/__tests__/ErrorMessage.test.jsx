/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import ErrorMessage from '../ErrorMessage';

describe('ErrorMessage', () => {
  let props;

  beforeEach(() => {
    props = {
      message: '',
    };
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<ErrorMessage />);
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot with props', () => {
    props.message = 'Error message';
    const wrapper = shallow(<ErrorMessage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('has h1 element containing Error', () => {
    const wrapper = shallow(<ErrorMessage {...props} />);
    expect(wrapper.find('h1').text()).toMatch('Error');
  });

  it('has h2 element containing passed message prop', () => {
    props.message = 'Error message';
    const wrapper = shallow(<ErrorMessage {...props} />);
    expect(wrapper.find('h2').text()).toMatch('Error message');
  });
});
