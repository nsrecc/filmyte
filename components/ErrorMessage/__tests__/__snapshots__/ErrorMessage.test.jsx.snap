// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage matches snapshot 1`] = `
<div
  className="errorMessageWrapper"
>
  <h1>
    Error.
  </h1>
  <h2>
    Please try again later.
  </h2>
</div>
`;

exports[`ErrorMessage matches snapshot with props 1`] = `
<div
  className="errorMessageWrapper"
>
  <h1>
    Error.
  </h1>
  <h2>
    Error message. Please try again later.
  </h2>
</div>
`;
